### PR TITLE
Use 1 agent to run multiple baseline testplans

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205
- [x] 202305

#### What is the motivation for this PR?
Baseline test will run multiple testplans at the same time, if each testplan use 1 agent, will be a lot of cost, and not good to future upgrade
#### How did you do it?
Use 1 agent to run multiple testplans which has same topo

### CODE REVIEW

**Review by lerry-lee — COMMENTED** [review]
There is almost the same logic. Can we use the single `run-test-elastictest-template.yml` for the baseline test scenario?

**Comment by xwjiang-ms** [comment]
> There is almost the same logic. Can we use the single `run-test-elastictest-template.yml` for the baseline test scenario?

Fixed

File: .azure-pipelines/run-test-elastictest-template.yml [hunk 1]
```diff
@@ -197,27 +203,29 @@ steps:
...
+      TEST_PLAN_ID_LIST=( $(cat new_test_plan_id.txt) )
+      for TEST_PLAN_ID in "${TEST_PLAN_ID_LIST[@]}"
+      do
+          echo "Created test plan $TEST_PLAN_ID"
+          echo -e "\033[33mSONiC PR system-level test is powered by SONiC Elastictest, for any issue, please send email to sonicelastictest@microsoft.com \033[0m"
+          echo -e -n "\033[33mPlease visit Elastictest page \033[0m"
+          echo -n "$(FRONTEND_URL)/scheduler/testplan/$TEST_PLAN_ID "
+          echo -e "\033[33mfor detailed test plan progress \033[0m"
+          echo "##vso[task.setvariable variable=TEST_PLAN_ID]$TEST_PLAN_ID"
+      done
```
Comment (lerry-lee) [inline .azure-pipelines/run-test-elastictest-template.yml:228]: What's the output if create multiple test plans. If it was not easy to watch, can we group each test plan creation message?

Comment (xwjiang-ms) [reply .azure-pipelines/run-test-elastictest-template.yml]: agree, SONiC PR system-level msg can only print once

**Review by lerry-lee — APPROVED** [review]

File: .azure-pipelines/run-test-elastictest-template.yml [hunk 2]
```diff
@@ -138,13 +138,15 @@ parameters:
     type: string
     default: ""
 
+  - name: TEST_PLAN_NUM
```
Comment (yejianquan) [inline .azure-pipelines/run-test-elastictest-template.yml:141]: Default value should be assigned in test_plan.py
Let's keep empty value in the templates

File: .azure-pipelines/baseline_test/baseline.test.template.yml [hunk 3]
```diff
@@ -120,14 +113,13 @@ jobs:
         BUILD_REASON: "BaselineTest"
         RETRY_TIMES: "0"
         STOP_ON_FAILURE: "False"
+        TEST_PLAN_NUM: $(BASELINE_MGMT_PUBLIC_MASTER_TEST_NUM)
```
Comment (yejianquan) [inline .azure-pipelines/baseline_test/baseline.test.template.yml:116]: Please confirm that other feature branches like 202305 and nightly test plan would be affected by the change of .azure-pipelines/run-test-elastictest-template.yml

File: .azure-pipelines/run-test-elastictest-template.yml [hunk 4]
```diff
@@ -138,13 +138,16 @@ parameters:
...
+    default: "1"
 
 steps:
   - ${{ if not(contains(variables['BUILD.REPOSITORY.NAME'], 'sonic-mgmt')) }}:
       - script: |
-          # If not sonic-mgmt/sonic-mgmt-int repo, need to download test_plan.py and pr_test_scripts.yaml
+          # If not sonic-mgmt/sonic-mgmt-int repo, need to download handle_test_plan.py, test_plan.py and pr_test_scripts.yaml
           set -ex
-
+          curl "https://raw.githubusercontent.com/sonic-net/sonic-mgmt/master/.azure-pipelines/handle_test_plan.py" -o ./.azure-pipelines/handle_test_plan.py
```
[the remark this answers is not in the document]
Comment (xwjiang-ms) [reply .azure-pipelines/run-test-elastictest-template.yml]: fixed

File: .azure-pipelines/run-test-elastictest-template.yml [hunk 5]
```diff
@@ -197,27 +203,29 @@ steps:
       --features "${{ parameters.FEATURES }}" \
       --scripts-exclude "${{ parameters.SCRIPTS_EXCLUDE }}" \
       --features-exclude "${{ parameters.FEATURES_EXCLUDE }}" \
-      --specific-param='${{ parameters.SPECIFIC_PARAM }}' \
-      --affinity='${{ parameters.AFFINITY }}' \
+      --specific-param="${{ parameters.SPECIFIC_PARAM }}" \
```
[the remark this answers is not in the document]
Comment (xwjiang-ms) [reply .azure-pipelines/run-test-elastictest-template.yml]: fixed

File: .azure-pipelines/run-test-elastictest-template.yml (hunk 2 above)
Comment (xwjiang-ms) [reply .azure-pipelines/run-test-elastictest-template.yml:141]: fixed

File: .azure-pipelines/baseline_test/baseline.test.template.yml (hunk 3 above)
Comment (xwjiang-ms) [reply .azure-pipelines/baseline_test/baseline.test.template.yml:116]: confirmed, with the change, 202305 and nightly test plan could work well

**Review by yejianquan — APPROVED** [review]
LGTM

**Pull request merged**